### PR TITLE
Laplacian filter

### PR DESCRIPTION
**Pull request by khilanravani**
The Laplacian of an image highlights regions of rapid intensity change and is therefore often used for edge detection. Unlike the Sobel edge detector, the Laplacian edge detector uses only one kernel. It calculates second order derivatives in a single pass.

### CODE REVIEW

**Comment by khilanravani** [comment]
Sir, here is the output of the laplacian filter:
laplacian_output

File: src/Graphics/Image/Processing/Filter.hs [hunk 1]
```diff
@@ -140,3 +141,31 @@ prewittOperator !img = sqrt (prewittX ^ (2 :: Int) + prewittY ^ (2 :: Int))
...
+-- >>> writeImage "images/yield_laplacian.png" (toImageRGB laplacianImage)
+--
+laplacianFilter :: (Array arr cs e, Array arr X e) =>
+                   Direction -> Border (Pixel cs e) -> Filter arr cs e
+laplacianFilter dir !border =
+  Filter (correlate border kernel)
+  where
+    !kernel =
+      case dir of
+        Complete   -> fromLists $ [ [ -1, -1, -1 ]     -- Unlike the Sobel edge detector, the Laplacian edge detector uses only one kernel.
```
Comment (alpmestan) [inline src/Graphics/Image/Processing/Filter.hs]: It's a bit annoying that nothing but `Complete` can be handled by `laplacianFilter`, and conversely the filters who use the other `Direction` constructors will not be able to handle `Complete`. So there's always a risk of runtime error if you add `Complete` there. Also, what is the point of doing it if that's the only option that `laplacianFilter` can handle anyway?

File: src/Graphics/Image/Processing/Gabor.hs [hunk 2]
```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE RankNTypes #-}
```
Comment (alpmestan) [inline src/Graphics/Image/Processing/Gabor.hs]: I suppose this file should not be part of this PR but it just slipped in?

File: hip.cabal [hunk 3]
```diff
@@ -70,6 +70,7 @@ Library
                  , Graphics.Image.Processing.Filter
                  , Graphics.Image.Processing.Hough
                  , Graphics.Image.Processing.Ahe
+                 , Graphics.Image.Processing.Laplacian
```
Comment (lehins) [inline hip.cabal]: There is no module with that name included in this PR

Comment (lehins) [reply hip.cabal]: @khilanravani If you'd like to place filters in their own modules you could do that, eg. a laplacian could go in `Graphics.Image.Processing.Filter.Laplacian`, but IMHO there is no real need for that. It's up to you though.

**Comment by lehins** [comment]
@alpmestan Thank you for your feedback.
@khilanravani Good work. Once you'll address the comments in the PRs we can go ahead and merge them. Oh and, I didn't look through Gabor filter, since it's better to have a separate PR for that.

File: src/Graphics/Image/Processing/Gabor.hs (hunk 2 above)
Comment (khilanravani) [reply src/Graphics/Image/Processing/Gabor.hs]: Yes sir, very sorry. Will correct it.

**Comment by khilanravani** [comment]
@lehins @alpmestan Thank u so much sir! Yes sir, I'll start a separate PR for Gabor soon (It's incomplete at the moment, got added in this branch by mistake :sweat_smile: ). Also, will make the changes and push them soon...

File: hip.cabal (hunk 3 above)
Comment (khilanravani) [reply hip.cabal]: Yes sir, will correct it.

File: src/Graphics/Image/Processing/Filter.hs (hunk 1 above)
Comment (khilanravani) [reply src/Graphics/Image/Processing/Filter.hs]: Ok sir, will do so.

**Comment by khilanravani** [comment]
@lehins Sir, I've made the changes. Kindly review them.
